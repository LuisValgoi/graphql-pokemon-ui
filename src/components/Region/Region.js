import React from 'react';

import { useIsSmallScreen } from '../../hooks/useIsMobile';

export const RegionTitle = (props) => {
  const { children, style } = props;
  const region = {
    wrapper: {
      border: '.5rem solid #F7F7F7'
    },
    title: {
      padding: '.5rem',
      color: '#007bff',
      border: '1px solid rgba(0,0,0,.125)',
      backgroundColor: 'white',
      borderRadius: 'calc(.25rem - 1px) calc(.25rem - 1px) calc(.25rem - 1px) calc(.25rem - 1px)',
      verticalAlign: 'middle',
      textAlign: 'center'
    }
  }

  return (
    <div style={{ ...region.wrapper, ...style }}>
      <div style={region.title}>
        {children}
      </div>
    </div >
  );
};

export const RegionImage = (props) => {
  const { children, style } = props;
  const region = {
    wrapper: {
      border: '.5rem solid #F7F7F7',
      backgroundColor: 'white',
      verticalAlign: 'middle',
      textAlign: 'center'
    },
    image: {
      border: '1px solid rgba(0,0,0,.125)',
      borderRadius: 'calc(.25rem - 1px) calc(.25rem - 1px) calc(.25rem - 1px) calc(.25rem - 1px)'
    }
  };

  return (
    <div style={{ ...region.wrapper, ...style }}>
      <div style={region.image}>
        {children}
      </div>
    </div>
  );
}

export const RegionContent = (props) => {
  const { children, style } = props;
  const region = {
    wrapper: {
      border: '.5rem solid #F7F7F7'
    },
    info: {
      border: '1px solid rgba(0,0,0,.125)',
      backgroundColor: 'white',
      borderRadius: 'calc(.25rem - 1px) calc(.25rem - 1px) calc(.25rem - 1px) calc(.25rem - 1px)',
      padding: '.5rem'
    }
  }

  return (
    <div style={{ ...region.wrapper, ...style }}>
      <div style={region.info}>
        {children}
      </div>
    </div >
  );
};

export const Region = (props) => {
  const isSmallScreen = useIsSmallScreen();
  const { children } = props;
  const region = {
    wrapper: {
      marginTop: isSmallScreen ? '3rem' : '1rem',
      padding: '.1rem',
      border: '1px solid rgba(0,0,0,.125)',
      borderRadius: 'calc(.25rem - 1px) calc(.25rem - 1px) calc(.25rem - 1px) calc(.25rem - 1px)',
      backgroundColor: '#F7F7F7'
    }
  };
  return (
    <div style={region.wrapper}>
      {children}
    </div>
  )
};
