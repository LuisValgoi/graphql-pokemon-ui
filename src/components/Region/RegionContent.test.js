import React from 'react';

import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { RegionContent } from './Region';

test('RegionContent: should render', () => {
  const { asFragment } = render(<RegionContent />)
  expect(asFragment()).toMatchSnapshot();
});

test('RegionContent: should render element passed as children using the given structure', () => {
  render(<RegionContent>text</RegionContent>);

  const content = screen.getByTestId('region-content-wrapper');
  const info = screen.getByTestId('region-content-info');
  expect(content).toContainElement(info);
  expect(info).toContainHTML('text');
});
