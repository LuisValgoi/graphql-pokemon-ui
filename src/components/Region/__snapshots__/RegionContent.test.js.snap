// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegionContent: should render 1`] = `
<DocumentFragment>
  <div
    data-testid="region-content-wrapper"
    style="border: .5rem solid #f7f7f7;"
  >
    <div
      data-testid="region-content-info"
      style="background-color: white; border-radius: calc(.25rem - 1px) calc(.25rem - 1px) calc(.25rem - 1px) calc(.25rem - 1px); padding: .5rem;"
    />
  </div>
</DocumentFragment>
`;
